src/Tasks/Microsoft.NET.Build.Tasks/ResolvePublishAssemblies.cs

```diff
@@ -45,12 +47,20 @@ protected override void ExecuteCore()
             IEnumerable<string> privateAssetsPackageIds = PackageReferenceConverter.GetPackageIds(PrivateAssetsPackageReferences);
             IPackageResolver packageResolver = NuGetPackageResolver.CreateResolver(lockFile, ProjectPath);
 
+            LockFile FilterLockFile = null;
+            if (!string.IsNullOrEmpty(FilterProjectAssetsFile))
+            {
+                FilterLockFile = new LockFileCache(BuildEngine4).GetLockFile(FilterProjectAssetsFile);
```

Comment:
Correct me if i am missing something, i thought LockFileCache maps to a single asset file. We are dealing with two asset files here 
AssetsFilePath - the App project
FilterLockFile  - the Filter profile supplied